cmd/operator-sdk/new/cmd.go

```diff
@@ -140,28 +115,46 @@ var (
 	makeVendor       bool
 	skipValidation   bool
 	generatePlaybook bool
+	flagSet          apiflags.AddAPIFlags
 
 	helmChartRef     string
 	helmChartVersion string
 	helmChartRepo    string
 
 	crdVersion string
+)*/
+
+var (
+	fs          apiflags.AddAPIFlags
+	projectName string
+	repo        string
 )
 
 func newFunc(cmd *cobra.Command, args []string) error {
 	if err := parse(cmd, args); err != nil {
 		return err
 	}
 	mustBeNewProject()
-	if err := verifyFlags(); err != nil {
+
+	// VerifyFlags
+	if err := fs.VerifyFlags(); err != nil {
 		return err
 	}
+	if fs.OperatorType == projutil.OperatorTypeGo {
+		if len(fs.APIVersion) != 0 || len(fs.Kind) != 0 {
+			return fmt.Errorf("operators of type Go do not use --api-version or --kind")
```

Comment:
Why not reduce the complexity here and allow always scaffold for all types when the flags are informed in the new?

Reply:
@joelanford 
Improvement: As far as I understand, This would fall under improving the current implementation.

Reply:
If you think that it will increase the scope then it is fine. My suggestion here is because in my mind would be easier just make all use the same impl. I mean instead of add the check just:

- > keep the same impl to create that already exist
- > encapsulate the api impl in methods per type as suggested above 
- > then here we just need to do a switch case with the operator type calling the api func for each type.

Reply:
Right, Above check is a very specific one just for the `Go` and only is needed for `new` command. Rest all of the Flags verification has been moved under /internal/flags/apiflags/flags.go

Reply:
I think we should go ahead and fix this one. I think you can move the things from line 144-150 and incorporate them into the switch/case statement:
```
        if len(fs.APIVersion) != 0 || len(fs.Kind) != 0 { 
            return fmt.Errorf("operators of type Go do not use --api-version or --kind")
        }   

        if err := projutil.CheckRepo(fs.Repo); err != nil {
            return err 
        }   
```

That will reduce the "complexity" from a linter point of view.

Reply:
Kubebuilder doesn't work this way, so I'd rather we not change functionality here then immediately change it back when we switch to KB.

Reply:
i.e. Kubebuilder `init` and `create api` are separate. I would actually be interested in talking through whether we should force Ansible and Helm operators to use the same workflow for consistency.

Reply:
I think helm and ansible ideally should have a similar workflow